src/native/libs/System.Native/pal_uid.c

```diff
@@ -237,3 +238,83 @@ int32_t SystemNative_GetGroups(int32_t ngroups, uint32_t* groups)
 
     return getgroups(ngroups, groups);
 }
+
+char* SystemNative_GetGroupName(uint32_t gid)
+{
+#if HAVE_GETGRGID_R
```

Comment:
According to https://cs.android.com/android/platform/superproject/+/master:bionic/docs/status.md, `getgrgid_r` is supported as of API level 24. But my understanding is that we are only building a single `.so` file, and targeting API level 21.

Would it make sense to try dynamically loading `getgrgid_r` on Android, and only fail if it can't be found at runtime?

cc @lambdageek @akoeplinger

Reply:
We could but in mono we just fell back to using `getgrgid`. If you look at the bionic implementation they end up doing the same thing anyway so it is already "thread-safe":
- https://github.com/aosp-mirror/platform_bionic/blob/master/libc/bionic/grp_pwd.cpp#L764
- https://github.com/aosp-mirror/platform_bionic/blob/master/libc/bionic/grp_pwd.cpp#L715

I'd use `getgrgid` and leave a TODO to switch over to the other API once we're targetting Android 24+